src/sign/index.ts

```diff
@@ -107,9 +107,11 @@ export default class Client {
       fetch(address, init)
         .then((res) => {
           if (res.ok) return resolve(res.json());
+          if (res.headers.get('content-type')?.includes('application/json'))
+            return res.json().then(reject).catch(reject);
           throw res;
         })
-        .catch((e) => reject(e));
+        .catch(reject);
```

Comment:
This will reject with an Error object, instead of json, no ? Same with line 111.

What's the goal of this PR ? To always return a JSON object when rejecting ?

Reply:
> This will reject with an Error object, instead of json, no ? Same with line 111.

Correct, this is for unknown errors like network error
for example `TypeError: Network request failed` 

we can still return object in this case also, but this will help us debug much better